loadtest/control/simplecontroller/controller.go

```diff
@@ -87,8 +87,8 @@ func (c *SimpleController) Run() {
 	cycleCount := 1 // keeps a track of how many times the entire cycle of actions have been completed.
 	for {
 		for i := 0; i < len(c.actions); i++ {
-			if cycleCount%c.actions[i].runFrequency == 0 {
-				// run the action if runFrequency is not set, or else it's set and it's a multiple
+			if cycleCount%c.actions[i].runPeriod == 0 {
```

Comment:
This is unrelated but since we are here can we check that `runPeriod` is not zero? Otherwise we will crash dividing by zero. The other option is to not accept zero as a valid value during parsing but I think it's fine to leave it as it is since it might be a quicker way to temporarily disable some action without removing it from the config.

Reply:
Good point, I am on the side that we should add this as a validation step.

Reply:
So you are suggesting to treat as not valid actions with a period equal to zero and fail the config validation if any is found?

What about logging a warning? We can skip those actions, avoid to add them in the final slice of actions to be executed but still go through with the validation.

In any case, if it comes to change the config validation code I suggest to make a new PR.

Reply:
I think we shouldn't continue with a invalid config. We need to be sure what value is going inside the main loop, validating in the loop (checking if zero here) doesn't sound right to me. Also if we add a warning here, depending on the speed it will cause some verbosity.

Reply:
The warning would only happen on config validation. Those actions with a period of zero will never reach the loop.

https://github.com/mattermost/mattermost-load-test-ng/blob/4e784afb1e2b742c24f31fe4f4b01d7e5fb7d82d/loadtest/control/simplecontroller/controller.go#L175-L186

We would only do the append if `def.RunPeriod` is greater than zero and warn if it's zero (which means we skip that action).

Reply:
What if we change the semantics for runPeriod to not run an action if it is 0 ? That would satisfy Claudio's point, and then we can add a config validation to fail any value < 0.

Reply:
Discarding the action definition if `runPeriod=0`. So, we are ok with graceful degradation :) Fine. I will add this change 👍

Reply:
Well, my first proposal was to simply allow 0 to be a valid value meaning "never run it".

Discarding the action is a different approach but I wouldn't call it "graceful degradation" since the overall outcome is the same as if those actions were not defined in the first place. It's just being less strict (more robust?) on the range of input and ignore invalid cases. 

To be perfectly honest, this is only needed because JSON doesn't have comments.
The better question would probably be why do we use that format for configuration but I guess that's a discussion for another day :stuck_out_tongue:

Reply:
> To be perfectly honest, this is only needed because JSON doesn't have comments.

Yep.